doc/developing/misc.md

```diff
@@ -36,3 +36,54 @@ Give both files to the developer and also provide:
 * commit the source was built from and/or modified source code (if applicable)
 
 Hotspot will resolve symbols correctly when pointing the sysroot option at the collected libs.
+
+
+## Profiling with Tracy
+
+[Tracy](https://github.com/wolfpld/tracy) is
+> A real time, nanosecond resolution, remote telemetry, hybrid frame and sampling
+> profiler for games and other applications.
+
+It allows one to annotate important functions and generate traces, where one can
+see when each individual function call happened, and how long it took.
+
+Tracy can also record when frames, e.g. server step, start and end, and inspect
+frames that took longer than usual. Minetest already contains annotations for
+its frames.
+
+See also [Tracy's official documentation](https://github.com/wolfpld/tracy/releases/latest/download/tracy.pdf).
+
+### Installing
+
+Tracy consists of a client (Minetest) and a server (the gui).
+
+Install the server, e.g. using your package manager.
+
+### Building
+
+Build Minetest with `-DDBUILD_WITH_TRACY=1`, this will fetch Tracy for building
+the Tracy client. And use `FETCH_TRACY_GIT_TAG` to get a version matching your
+Tracy server, e.g. `-DFETCH_TRACY_GIT_TAG=v0.11.0` if it's `0.11.0`.
+
+To actually use Tracy, you also have to enable it with Tracy's build options:
+```
+-DTRACY_ENABLE=1 -DTRACY_ONLY_LOCALHOST=1
```

Comment:
maybe we should set these by default?

Reply:
Not necessary imo.

Also, idk how to do this properly in cmake. x)